### PR TITLE
PORT 8756 aws integration add use get resource api to docs

### PR DESCRIPTION
**Pull request by shalev007**
# Description

- **Add `useGetResourceAPI` option to the AWS Lambda function mapping in the `mapping_extra_resources.md`.**
- **feat: add helm installation**
- **feat: add installation warnings**

## Updated docs pages

- AWS Installation (/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation)
- Mapping extra resources (/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/examples/mapping_extra_resources)

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 1]
```diff
@@ -38,12 +44,35 @@ The AWS integration uses the following AWS infrastructure:
 
 In order to successfully deploy the AWS integration, it's crucial to ensure that the user who deploys the integration in the AWS Organization has the appropriate access permissions to create all of the above resources.
 
-## Installation walkthrough using terraform
+## install using terraform
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:47]: The first word in headers should be capitalized (see other headers as well)

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/examples/mapping_extra_resources.md [hunk 2]
```diff
@@ -162,13 +162,54 @@ Create an integration configuration for the resource. The integration configurat
                   # highlight-end
             ```
 
-    :::tip
-    To get an example of the AWS resource properties, you can use the [AWS Cloud Control API](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/what-is-cloudcontrolapi.html) to get the resource properties.
+#### `useGetResourceAPI` property
 
-    For example for the `AWS::Lambda::Function` resource, you can use the following command to get the resource properties:
+- By default the integration uses the [`CloudControl:ListResources`](https://docs.aws.amazon.com/cli/latest/reference/cloudcontrol/list-resources.html) API to get the resources. The integration can also enrich each resource by running [`CloudControl:GetResource`](https://docs.aws.amazon.com/cli/latest/reference/cloudcontrol/get-resource.html) on each resource, you can use this by enabling `useGetResourceAPI` option.
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/examples/mapping_extra_resources.md:167]:
```suggestion
- By default the integration uses the [`CloudControl:ListResources`](https://docs.aws.amazon.com/cli/latest/reference/cloudcontrol/list-resources.html) API to get the resources. The integration can also enrich each resource by running [`CloudControl:GetResource`](https://docs.aws.amazon.com/cli/latest/reference/cloudcontrol/get-resource.html) on each resource, you can use this by enabling `useGetResourceAPI` option.  

```

**Review by hadar-co — CHANGES_REQUESTED** [review]

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/examples/mapping_extra_resources.md [hunk 3]
```diff
@@ -162,13 +162,55 @@ Create an integration configuration for the resource. The integration configurat
...
+            tags: '.Properties.Tags'
+            iamRole: "'https://console.aws.amazon.com/go/view?arn=' + .Properties.Role"
+            arn: '.Properties.Arn'
+          relations:
+            account: '.__AccountId'
+```
+
+**Note: Using the `useGetResourceAPI` option will make each resync run slower.**
+
+:::tip Get an example of the AWS resource properties
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/examples/mapping_extra_resources.md:207]:
```suggestion
:::tip AWS resource properties example
```

**Review by hadar-co — APPROVED** [review]

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 4]
```diff
@@ -7,6 +7,12 @@ sidebar_position: 1
 The AWS integration is deployed using Terraform on AWS ECS cluster service.  
 It uses our Terraform [Ocean](https://ocean.getport.io) Integration Factory [module](https://registry.terraform.io/modules/port-labs/integration-factory/ocean/latest) to deploy the integration.
 
+:::warning Live-Events
+This installation guide is for the AWS integration only.
+It does not take into consideration the **live-events** infrastructure **which is optional**.
+The env variables referring to the live events (such as `LIVE_EVENTS_API_KEY`) are optional and can be removed if not needed.
+:::
+
 ## Infrastructure
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:17]: should be under the tab of the real time events / terraform

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 5]
```diff
@@ -2,11 +2,119 @@
...
+<Tabs groupId="installation-platforms" queryString="installation-platforms">
+<TabItem value="helm" label="Helm">
+The AWS integration is deployed using Helm on you cluster.
+You can check out the Helm chart [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-ocean).
+
+## Prerequisites
+
+- [Helm 3](https://helm.sh/docs/intro/install/)
+- [A logged in aws CLI 2](https://aws.amazon.com/cli/)
+
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:24]: Write about the PORT_CLIENT_ID and secret, and aws key id and secret and where to get it

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 6]
```diff
@@ -2,11 +2,119 @@
...
+helm upgrade --install aws port-labs/port-ocean \
+--set port.clientId="$PORT_CLIENT_ID"  \
+--set port.clientSecret="$PORT_CLIENT_SECRET_ID"  \
+--set port.baseUrl="https://api.getport.io"  \
+--set initializePortResources=true  \
+--set sendRawDataExamples=true  \
+--set integration.identifier="my-aws"  \
+--set integration.type="aws"  \
+--set integration.eventListener.type="POLLING"  \
+--set integration.config.liveEventsApiKey="$YOUR_CUSTOM_API_KEY" \
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:36]: remove

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 7]
```diff
@@ -2,11 +2,119 @@
...
+--set integration.config.awsSecretAccessKey="$AWS_SECRET_ACCESS_KEY" \
+```
+
+### IRSA
+
+If you are using [IRSA](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html), you can set the following values, after [assigning the appropriate IAM role to the service account](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html):
+
+```bash
+--set podServiceAccount.name="$SERVICE_ACCOUNT" \
+```
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:47]: show full installation method

Comment (Tankilevitch) [reply docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:47]: remove `\`

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 8]
```diff
@@ -2,11 +2,119 @@
...
+--set port.clientSecret="$PORT_CLIENT_SECRET_ID"  \
+--set port.baseUrl="https://api.getport.io"  \
+--set initializePortResources=true  \
+--set sendRawDataExamples=true  \
+--set integration.identifier="my-aws"  \
+--set integration.type="aws"  \
+--set integration.eventListener.type="POLLING"  \
+--set integration.config.liveEventsApiKey="$YOUR_CUSTOM_API_KEY" \
+--set integration.config.awsAccessKeyId="$AWS_ACCESS_KEY_ID" \
+--set integration.config.awsSecretAccessKey="$AWS_SECRET_ACCESS_KEY" \
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:38]: remove `\`

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 9]
```diff
@@ -2,11 +2,119 @@
...
+In order to successfully deploy the AWS integration, it's crucial to ensure that the user who deploys the integration in the AWS Organization has the appropriate access permissions to create all of the above resources.
+
+Choose one of the following installation methods:
+<Tabs groupId="installation-platforms" queryString="installation-platforms">
+<TabItem value="helm" label="Helm">
+The AWS integration is deployed using Helm on you cluster.
+You can check out the Helm chart [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-ocean).
+
+## Prerequisites
+
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:21]: missing information about how to create the user and get the access key and secret
and what roles to assign to it

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md [hunk 10]
```diff
@@ -26,72 +134,36 @@ The AWS integration uses the following AWS infrastructure:
...
+| `OCEAN__EVENT_LISTENER`                              | [The event listener object](https://ocean.getport.io/framework/features/event-listener/).                                                                                                                                                                            |
+| `OCEAN__INTEGRATION__IDENTIFIER`                     | The identifier of the integration.                                                                                                                                                                                                                                   |
+| `OCEAN__INTEGRATION__TYPE`                           | should be set to `aws`.                                                                                                                                                                                                                                              |
+
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:163]: missing installation example for on-premise

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md (hunk 10 above)
Comment (Tankilevitch) [reply docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:163]: also lets don't show the live key events parameter in the docker example

Comment (Tankilevitch) [reply docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/aws/installation.md:163]: and lets describe what properties needs to be filled by the user in each of the installation methods

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**